core/tests/window/test_window.py

```diff
@@ -99,6 +99,29 @@ def test_set_app_with_content(window, app):
     assert content.app == app
 
 
+def test_set_app_with_content_at_instantiation(app):
+    """A window can be created with content"""
+    try:
+        window_no_content = toga.Window()
+        assert window_no_content.content is None
+    finally:
+        window_no_content.close()
+        window_no_content = None
+
+    # Setup the box with something inside it:
+    label1 = toga.Label("Hello World")
+    content = toga.Box(children=[label1])
+    try:
+        window_with_content = toga.Window(content=content)
+        assert window_with_content.content == content
+        # Make sure it still has that internal content:
+        assert len(window_with_content.content.children) == 1
+        assert window_with_content.content.children[0] == label1
+    finally:
+        window_with_content.close()
+        window_with_content = None
```

Comment:
There's no need to do cleanup like this in the dummy backend - it doesn't leave any artefacts that could be a problem.